contrib_bots/test-bots

```diff
@@ -30,14 +31,26 @@ if __name__ == '__main__':
 
     sys.path.insert(0, root_dir)
 
-    # mypy doesn't recognize the TestLoader attribute, even though the code
-    # is executable
-    loader = unittest.TestLoader() # type: ignore
-    if args.bot is not None:
-        bots_test_dir = dir_join(bots_test_dir, args.bot[0])
-    suite = loader.discover(start_dir=bots_test_dir, top_level_dir=root_dir)
-    runner = unittest.TextTestRunner(verbosity=2)
-    # same issue as for TestLoader
-    result = runner.run(suite) # type: ignore
-    if result.errors or result.failures:
-        raise Exception('Test failed!')
+    def run_tests_in(start_dir):
+        # type: (str) -> None
+        # mypy doesn't recognize the TestLoader attribute, even though the code
+        # is executable
+        loader = unittest.TestLoader() # type: ignore
+        suite = loader.discover(start_dir=start_dir, top_level_dir=root_dir)
+        runner = unittest.TextTestRunner(verbosity=2)
+        # same issue as for TestLoader
+        result = runner.run(suite) # type: ignore
+        if result.errors or result.failures:
+            raise Exception('Test failed!')
+
+    if args.only is not None:
+        bot_names = args.only[0].split(',')
+        btd = bots_test_dir
+        available_bots = [b for b in os.listdir(btd) if os.path.isdir(dir_join(btd, b))]
+        for n in bot_names:
+            if n not in available_bots:
+                logging.error("Bot with name '%s' is unavailable for testing." % (n))
+            else:
+                run_tests_in(dir_join(bots_test_dir, n))
+    else:
+        run_tests_in(bots_test_dir)
```

Comment:
I've added this now, so later tests attempt to run if an earlier one is incorrect/unavailable; the question is whether the names specified should be checked ASAP (as per the 'choices' option in argparse, as above, for example), or just as 'best attempt' is made to run what is selected?